File: easy/26.删除排序数组中的重复项.js
/*
 * @lc app=leetcode.cn id=26 lang=javascript
 *
 * [26] 删除排序数组中的重复项
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
  let x = 0;
  for (let i = 0; i < nums.length; i += 1) {
    if (nums[x] !== nums[i]) {
      nums[++x] = nums[i];
    }
  } 
  return ++x;
};
// @lc code=end

File: easy/26.删除排序数组中的重复项.py
#
# @lc app=leetcode.cn id=26 lang=python3
#
# [26] 删除排序数组中的重复项
# 返回移除后数组的新长度

# @lc code=start
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        x = 1
        for i in range(len(nums) - 1):
            if nums[i] != nums[i + 1]:
                nums[x] = nums[i + 1]
                x += 1
        return x

# @lc code=end

